tst/regression/test_suites/restart/restart.py

```diff
@@ -49,7 +49,9 @@ def Analyse(self, parameters):
         silverFile = 'silver.out0.00002.rhdf'
 
         gold = h5py.File(goldFile,'r')
+        gold.close()
         silver = h5py.File(silverFile,'r')
+        silver.close()
```

Comment:
I don't think this is right. You won't be able to read the data with this. What about the following?

```python
def Analyse(self, parameters):
        # spotcheck one variable
        goldFile = 'gold.out0.00002.rhdf'
        silverFile = 'silver.out0.00002.rhdf'
        varName = "/advected"

        with h5py.File(goldFile,'r') as gold:
                goldData = gold[varName][:].flatten()

        with h5py.File(silverFile,'r') as silver:
                silverData = gold[varName][:].flatten()

        # spot check on one variable
        maxdiff = max(abs(goldData-silverData))
        print('Variable: %s, diff=%g, N=%d'%(varName,maxdiff,len(goldData)))

        return (maxdiff == 0.0)
```

This is also more pythonic. I'm pretty sure that the `with` function automatically cleans up the `gold` and `silver` h5py objects so you don't need to call `close`.